src/wazuh_modules/agent_upgrade/wm_agent_upgrade_result_command.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Wazuh Module for Agent Upgrading
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 3, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "wm_agent_upgrade.h"
+#include "wm_agent_parsing.h"
+#include "wm_agent_upgrade_tasks.h"
+#include "wazuh_db/wdb.h"
+
+cJSON* wm_agent_process_upgrade_result_command(const cJSON* agents) {
+    cJSON* response = cJSON_CreateArray();
+    cJSON* agents_info = wdb_select_agents_version(agents);
+    char *agents_info_string = cJSON_Print(agents_info);
+    mtdebug1(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_RESULT_AGENT_INFO, agents_info_string);
+    os_free(agents_info_string);
+    // My version: __ossec_version
+    for(int i = 0; i < cJSON_GetArraySize(agents); i++) {
+        bool agent_matched = false;
+        int agent_id = cJSON_GetArrayItem(agents, i)->valueint;
+
+        for (int j=0; j < cJSON_GetArraySize(agents_info); j++) {
+            cJSON *agent_info = cJSON_GetArrayItem(agents_info, j);    
+            if (agent_id == cJSON_GetObjectItem(agent_info, "agent_id")->valueint) {
+                agent_matched = true;
+                // Match in agent
+                if (strcmp(cJSON_GetObjectItem(agent_info, "version")->valuestring, __ossec_name " " __ossec_version) == 0) {
+                    // Agent updated
+                    cJSON_AddItemToArray(response, wm_agent_parse_response_mesage(SUCCESS, "Agent is updated", &agent_id, NULL, "UPDATED"));
```

Comment:
You can map the strings `"UPDATED"`, `"OUTDATED"` and `"UPDATING"` like you do with the error codes.